src/main/java/com/gtnewhorizons/navigator/api/model/layers/UniversalInteractableRenderer.java

```diff
@@ -104,10 +104,15 @@ public boolean onKeyPressed(int keyCode) {
             }
         }
 
-        if (Util.isKeyPressed(NavigatorApi.ACTION_KEY) && hoveredRenderStep != null) {
-            hoveredRenderStep.onActionKeyPressed();
-            manager.forceRefresh();
-            return true;
+        if (hoveredRenderStep != null) {
+            if (Util.isKeyPressed(NavigatorApi.ACTION_KEY)) {
+                hoveredRenderStep.onActionKeyPressed();
+                manager.forceRefresh();
+                return true;
+            } else if (hoveredRenderStep.onActionWithKeyPressed(keyCode)) {
+                manager.forceRefresh();
+                return true;
+            }
```

Comment:
If you follow the other example then all of these can be turned into

```suggestion
       if (hoveredRenderStep != null && hoveredRenderStep.onKeyPressed(keyCode)) {
            manager.forceRefresh();
            return true;
        }
```

Reply:
Fixed